lib/jsonapi/serializer.ex

```diff
@@ -163,4 +163,20 @@ defmodule JSONAPI.Serializer do
 
   def get_view({view, :include}), do: view
   def get_view(view), do: view
+
+  def maybe_dasherize_key(data) do
```

Comment:
Can we DRY these methods up at all?  If we remove `underscore_key/1` in favor of pattern matching in the Util module we might be able to.

I think we should change `dasherize` to `underscore` so that the code matches up to the configuration it's dealing with.  Also, is the maybe prefix necessary?

Reply:
I used `dasherize` as it felt like it better described what's happening to the data, and `maybe_` is a pattern I've seen used to indicate something may or may not happen.  I've replaced `maybe_dasherize` with `underscore` for consistency with the configuration.